internal/pkg/pluginengine/change.go

```diff
@@ -169,3 +182,41 @@ func handleResult(smgr statemanager.Manager, change *Change) error {
 	log.Successf("Plugin %s(%s) %s done.", change.Tool.Name, change.Tool.Plugin.Kind, change.ActionName)
 	return nil
 }
+
+// FillInputParams fill inputs from state
+func FillInputParams(smgr statemanager.Manager, options map[string]interface{}) error {
```

Comment:
This function doesn't need to be "exported", because it's only being used within the same package.

Reply:
I might also suggest a better name. "Fill input params" doesn't describe what it does exactly. handle outputs, fill outputs? Not very good examples, please think about it more.

Reply:
This is an important function that should be covered by unit tests.